src/main/java/ch/njol/skript/expressions/ExprSpawner.java

```diff
@@ -0,0 +1,131 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+
+package ch.njol.skript.expressions;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.bukkit.Material;
+import org.bukkit.block.Block;
+import org.bukkit.block.CreatureSpawner;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.classes.Converter;
+import ch.njol.skript.aliases.Aliases;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.entity.EntityType;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Spawner Type")
+@Description("Retrieves, sets, or resets the spawner's entity type")
+@Examples({"broadcast \"%spawner's entity type%\""})
+@Since("INSERT VERSION")
+public class ExprSpawner extends SimplePropertyExpression<Block, EntityData> {
+	
+	private static final Material MATERIAL_SPAWNER = Aliases.javaItemType("spawner").getMaterial();
+	private static final Map<EntityType, org.bukkit.entity.EntityType> CACHE = new HashMap<EntityType, org.bukkit.entity.EntityType>();
```

Comment:
If all keys are unique, you could use Guava's BiMap and then be able to use this cache for the other method as well.

Also, shouldn't the keys be EntityData, EntityType?

Reply:
EntityData for whatever reason causes a CCE because Skript's EntityType can't be cast to Skript's EntityData, so I just stuck to EntityData front-end but backend it uses EntityType

Reply:
Hm, that might be the same issue as #2347, I'll have to investigate further. But if it works, I don't have any issues with it.

Reply:
Yeah, I tested it out earlier, should work correctly with most entities* (Some Bukkit EntityTypes don't have a proper representation in Skript, so they're just paired with null, e.g. ``THROWN_EXP_BOTTLE``, but that shouldn't be too big of a deal right?

Reply:
Just discard the idea of the BiMap, almost all entities are an instance of SimpleEntityData and that may cause issues we don't really want to adress.